app/questionnaire/rules.py

```
@@ -30,20 +30,25 @@ def evaluate_goto(goto_rule, metadata, answer_store, group_instance):
     :return:
     """
     if 'when' in goto_rule.keys():
-        when = goto_rule['when']
+        all_conditions_met = True
 
-        if 'id' in when:
-            answer_index = when['id']
-            filtered = answer_store.filter(answer_id=answer_index, group_instance=group_instance)
-            if len(filtered) == 1:
-                return evaluate_rule(when, filtered[0]['value'])
+        for when_rule in goto_rule['when']:
+            condition_met = False
 
-        elif 'meta' in when:
-            key = when['meta']
-            value = get_metadata_value(metadata, key)
-            return evaluate_rule(when, value)
+            if 'id' in when_rule:
+                answer_index = when_rule['id']
+                filtered = answer_store.filter(answer_id=answer_index, group_instance=group_instance)
+                if len(filtered) == 1:
```

Should this raise an exception or at least log an error as an `else` to this; not sure what is expected if more than one answer comes back (can that happen for multiple answer_instances?) but currently it will silently return False which could be a nasty one to debug. Another solution could be to get answer_store.filter to take an `expected_match_count=1` and have it do the log/exception so anyone calling filter doesn't need to deal with it.

---

I think there are a few places in the navigator where we are in an erroneous state either where the schema may have been misconfigured or something we can't cope with has happened. Might be worth looking at the navigator as a whole to see if we should be raising errors in certain states

---

I don't think this is the responsibility of the navigator. We shouldn't have schemas that contain invalid routing rules, we should have better validation.

---

That's fine as long as when/if we do implementing routing based on multiple answers this gets picked up, it's easy to forget stuff like this and without any output indicating otherwise it can go undetected for a long time until someone finds a bug in production. Possible a perfect case for an `assert` statement actually.

---

I've updated this to log a warning and added a trello card to the backlog: https://trello.com/c/mXL1a8JD/297-update-navigator-to-handle-multiple-filtered-answers